src/Windows/Avalonia.Win32/ClipboardImpl.cs

```diff
@@ -106,9 +107,9 @@ public async Task<string[]> GetFormatsAsync()
 
                 if (hr == 0)
                 {
-                    var wrapper = new OleDataObject(dataObject);
+                    using var proxy = MicroCom.MicroComRuntime.CreateProxyFor<Win32Com.IDataObject>(dataObject, false);
```

Comment:
OleGetDataObject increments the reference counter of the data object. The proxy is created as a non-owning one, so data object will be never destroyed. That's a memory leak.

1) proxy should be created with ownsHandle: true
2) OleDataObject should call MicroComRuntime.CloneReference to obtain its own copy with its own reference since it retains it

Reply:
1. Thanks for explanation. Will change.
2. OleDataObject doesn't really lives more than "proxy" reference here. But I it does in another place, will do.